metricbeat/autodiscover/appender/kubernetes/token/token.go

```diff
@@ -41,7 +41,7 @@ type tokenAppender struct {
 // NewTokenAppender creates a token appender that can append a bearer token required to authenticate with
 // protected endpoints
 func NewTokenAppender(cfg *common.Config) (autodiscover.Appender, error) {
-	cfgwarn.Beta("The token appender is beta")
+	cfgwarn.Deprecate("7.0.0", "token appender is deprecated in favor of bearer_token_file config parameter")
```

Comment:
Do you suggest moving to `config` appender for use cases that require a bearer token then? I wouldnt do a blanket token path on all configs with the hints builder.

Reply:
Yeah, my thought was to use the `config` appender to set `bearer_token_file` wherever you were using the `token` appender before. No need to create new hints for this.
Would that work for you?

Reply:
im ok with that. we will do the needful once this is merged.